Sources/StreamChat/Models/Channel.swift

```diff
@@ -149,7 +151,7 @@ public struct _ChatChannel<ExtraData: ExtraDataTypes> {
         cooldownDuration: Int = 0,
         extraData: ExtraData.Channel,
 //        invitedMembers: Set<_ChatChannelMember<ExtraData.User>> = [],
-        latestMessages: [_ChatMessage<ExtraData>] = []
+        latestMessages: @escaping @autoclosure (() -> [_ChatMessage<ExtraData>]) = [],
```

Comment:
I must say I find this API quite confusing for the caller. For the optimization to work correctly, you need to know the fields are loaded lazily and you have to make sure you don't them prematurely. 

So, rather then `@escaping @autoclosure` it's IMO better and much more safer to have `() -> [_ChatMessage<ExtraData>]` explicitly here.